src/main/java/net/mcreator/ui/action/impl/workspace/DownloadJDK.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2021, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.action.impl.workspace;
+
+import net.mcreator.generator.GeneratorConfiguration;
+import net.mcreator.io.OS;
+import net.mcreator.io.UserFolderManager;
+import net.mcreator.io.zip.ZipIO;
+import org.apache.commons.io.FileUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.net.URL;
+
+public class DownloadJDK {

```

Comment:
A better name would be eg. JDKDownloader, even better if we called this SDK instead of JDK I think